itest/round_trip_send_test.go

```diff
@@ -0,0 +1,127 @@
+package itest
+
+import (
+	"context"
+
+	"github.com/lightninglabs/taro/tarorpc"
+	"github.com/lightningnetwork/lnd/lntest/wait"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+)
+
+// testFullValueSplitSend tests that we can properly send the full value of a
+// normal asset.
+func testRoundTripSend(t *harnessTest) {
+	// First, we'll make an normal assets with enough units to allow us to
+	// send it around a few times.
+	rpcAssets := mintAssetsConfirmBatch(
+		t, t.tarod, []*tarorpc.MintAssetRequest{simpleAssets[0]},
+	)
+
+	genInfo := rpcAssets[0].AssetGenesis
+	genBootstrap := rpcAssets[0].AssetGenesis.GenesisBootstrapInfo
+
+	ctxb := context.Background()
+
+	// Now that we have the asset created, we'll make a new node that'll
+	// serve as the node which'll receive the assets.
+	secondTarod := setupTarodHarness(
+		t.t, t, t.lndHarness.BackendCfg, t.lndHarness.Bob, t.universeServer,
+	)
+	defer func() {
+		require.NoError(t.t, secondTarod.stop(true))
+	}()
+
+	// We'll send half of the minted units to Bob, and then have Bob return
+	// half of the units he received.
+	fullAmt := rpcAssets[0].Amount
+	bobAmt := fullAmt / 2
+	aliceAmt := bobAmt / 2
+
+	// First, we'll send half of the units to Bob.
+	bobAddr, err := secondTarod.NewAddr(ctxb, &tarorpc.NewAddrRequest{
+		GenesisBootstrapInfo: genBootstrap,
+		Amt:                  bobAmt,
+	})
+	require.NoError(t.t, err)
+
+	assertAddrCreated(t.t, secondTarod, rpcAssets[0], bobAddr)
+	sendResp := sendAssetsToAddr(t, t.tarod, bobAddr)
+	sendRespJSON, err := formatProtoJSON(sendResp)
+	require.NoError(t.t, err)
+	t.Logf("Got response from sending assets: %v", sendRespJSON)
+
+	// Mine a block to force the send we created above to confirm.
+	_ = mineBlocks(t, t.lndHarness, 1, len(rpcAssets))
+
+	// Complete the send by exhanging proofs.
+	sendProof(t, t.tarod, secondTarod, bobAddr, genInfo)
```

Comment:
Perhaps we can wrap this and the mine+send above into a helper func? We'd then use the exact same func below when we send the assets back.